examples/scheduler/src/test/kotlin/cnt/scheduler/ProgrammaticSchedulerTest.kt

```diff
@@ -0,0 +1,67 @@
+package cnt.scheduler
+
+import io.mockk.called
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import io.quarkus.runtime.StartupEvent
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.InstanceOfAssertFactories
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.quartz.JobDetail
+import org.quartz.Scheduler
+import org.quartz.Trigger
+import org.quartz.impl.JobDetailImpl
+import org.quartz.impl.triggers.SimpleTriggerImpl
+import java.util.Date
+
+/**
+ * This is the preferred way to test the programmatically defined scheduler as we don't need to have a running quarkus context
+ * which needs time and resources.
+ */
+class ProgrammaticSchedulerTest {
+
+    @Nested
+    inner class AddSendBookNewsletterScheduler {
+
+        private val quartz: Scheduler = mockk()
+        private val sendReadingProbesNewsletterConfig: SendReadingProbesNewsletterConfig = mockk()
+        private val cut: ProgrammaticScheduler = ProgrammaticScheduler(quartz, sendReadingProbesNewsletterConfig)
+
+        private val event: StartupEvent = mockk()
+        private val jobDetailCaptor = slot<JobDetail>()
+        private val triggerCaptor = slot<Trigger>()
+
+        @Test
+        fun `scheduler is disabled - should not run`() {
+            every { sendReadingProbesNewsletterConfig.enabled() } returns false
+
+            cut.addSendBookNewsletterScheduler(event)
+
+            verify { quartz wasNot called }
+        }
+
+        @Test
+        fun `scheduler is enabled - should schedule job`() {
+            with(sendReadingProbesNewsletterConfig) {
+                every { enabled() } returns true
+                every { intervalInMilliseconds() } returns 30
+            }
+            every { quartz.scheduleJob(any(), any()) } returns Date()
+
+            cut.addSendBookNewsletterScheduler(event)
+
+            verify { quartz.scheduleJob(capture(jobDetailCaptor), capture(triggerCaptor)) }
+
+            assertThat(jobDetailCaptor.captured)
+                .asInstanceOf(InstanceOfAssertFactories.type(JobDetailImpl::class.java))
```

Comment:
:vomiting_face:

Reply:
😢

Reply:
While 🤮 is not very expressive, I assume what @Alexander-Miller wanted to say is:
"This looks a bit strange."

1. shouldn't the captor already know the type of what it captured?
2. the syntax is a bit hard to read

There is a nice feature from MockK where you can inline assert parameters within the verify block: `withArg {}`

So it should be possible write something like:
```
verify {
    quartz.scheduleJob(
        withArg { assertThat(it.name).isEqualTo(SEND_READING_PROBES_IDENTITY_SCHEDULER) },
        withArg { 
            assertThat(it.repeatCount).isEqualTo(-1)
            assertThat(it.name).isEqualTo(SEND_READING_PROBES_IDENTITY_TRIGGER)
        }
}
```